src/components/InventoryGroupDetail/GroupDetailHeader.js

```diff
@@ -21,20 +21,33 @@ import PropTypes from 'prop-types';
 import DeleteGroupModal from '../InventoryGroups/Modals/DeleteGroupModal';
 import RenameGroupModal from '../InventoryGroups/Modals/RenameGroupModal';
 import { fetchGroupDetail } from '../../store/inventory-actions';
+import { usePermissionsWithContext } from '@redhat-cloud-services/frontend-components-utilities/RBACHook';
+import {
+  REQUIRED_PERMISSIONS_TO_MODIFY_GROUP,
+  REQUIRED_PERMISSIONS_TO_READ_GROUP,
+} from '../../constants';
 
 const GroupDetailHeader = ({ groupId }) => {
   const dispatch = useDispatch();
   const { uninitialized, loading, data } = useSelector(
     (state) => state.groupDetail
   );
 
+  const { hasAccess: canRead } = usePermissionsWithContext(
+    REQUIRED_PERMISSIONS_TO_READ_GROUP(groupId)
+  );
+
+  const { hasAccess: canModify } = usePermissionsWithContext(
+    REQUIRED_PERMISSIONS_TO_MODIFY_GROUP(groupId)
+  );
+
   const [dropdownOpen, setDropdownOpen] = useState(false);
   const [renameModalOpen, setRenameModalOpen] = useState(false);
   const [deleteModalOpen, setDeleteModalOpen] = useState(false);
 
   const name = data?.results?.[0]?.name;
   const title =
-    uninitialized || loading ? (
+    canRead && (uninitialized || loading) ? (
```

Comment:
This seems a bit off. If canRead is falsy, I suspect the title will be equal to name or groupId?  I believe, one if-else would make this comparison easier to understand the logic even if it is really basic and not trendy.

Reply:
good catch, there was a bug, yes. revisited in the last commit